crates/interpreter/src/valid.rs

```diff
@@ -826,3 +898,67 @@ impl<'a, 'm> Expr<'a, 'm> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use alloc::format;
+
+    use proptest::prelude::*;
+
+    use super::bit_field::*;
+    use crate::Error;
+
+    proptest! {
+        #[test]
+        fn signed_field_round_trip(value in -8i32..8) {
+            let mask = 0b1111000;
+            let field = into_signed_field(mask, value).unwrap();
+            let recovered = from_signed_field(mask, field);
+            prop_assert_eq!(value, recovered);
```

Comment:
nit: let's use `actual` and `expected` vocabulary
```suggestion
        fn signed_field_round_trip(expected in -8i32..8) {
            let mask = 0b1111000;
            let actual = from_signed_field(mask, into_signed_field(mask, expected).unwrap());
            prop_assert_eq!(actual, expected);
```